book/bayesian_optimization.qmd

```diff
@@ -0,0 +1,844 @@
+  geom_stepribbon(aes(x = nr_eval, min = mean_y_min - se_y_min, max = mean_y_min + se_y_min, fill = method), colour = NA, alpha = 0.1, data = mean_results) +
+  labs(y = "Best Observed Function Value", x = "Nr. Function Evaluations", fill = "Method", colour = "Method") +
+  theme_minimal() +
+  theme(legend.position = "bottom")
+ggsave("Figures/bo_1d_sinusoidal_bo_rs.pdf", plot = g, width = 6, height = 4)
+ggsave("Figures/bo_1d_sinusoidal_bo_rs.png", plot = g, width = 6, height = 4)
+```
+
+```{r, echo=FALSE}
+#| echo: false
+#| label: fig-bayesian-sinusoidal_bo_rs
+#| fig-cap: Anytime performance of BO and random search (RS) on the 1D sinusoidal function given a budget of 20 function evaluations. Solid line depicts the best observed target value averaged over 10 replications. Ribbons represent standard errors.
+#| fig-alt: FIXME
+knitr::include_graphics("Figures/bo_1d_sinusoidal_bo_rs.png")
+```
+
+## Bayesian Optimization for Hyperparameter Optimization {#sec-bayesian-tuning}
+
+`r mlr3mbo` can be used out of the box for HPO (tuning) within the mlr3 ecosystem using a `r ref("TunerMbo")`.
+For illustrative purposes, we revisit the tuning example of @sec-tuning-instance and perform BO instead of a grid search.
+
+First, we construct the tuning instance:
+
+```{r}
+resampling = rsmp("cv", folds = 3)
+
+measure = msr("classif.ce")
+
+learner = lrn("classif.svm",
+  cost  = to_tune(1e-5, 1e5, logscale = TRUE),
+  gamma = to_tune(1e-5, 1e5, logscale = TRUE),
+  kernel = "radial",
+  type = "C-classification"
+)
+
+instance = ti(
+  task = tsk("sonar"),
+  learner = learner,
+  resampling = rsmp("cv", folds = 3),
+  measures = msr("classif.ce"),
+  terminator = trm("evals", n_evals = 25)
+)
+```
+
+We can then simply construct an `r ref("TunerMbo")` and use it to optimize the instance.
+Note that `r ref("TunerMbo")` is simply a light-weight wrapper around `r ref("OptimizerMbo")`.
+
+```{r}
+tuner = tnr("mbo",
+  loop_function = bayesopt_ego,
+  surrogate = surrogate,
+  acq_function = acq_function,
+  acq_optimizer = acq_optimizer)
+```
+
+```{r, output=FALSE}
+tuner$optimize(instance)
+```
+
+```{r}
+instance$result
+```
+
+We see that BO finds a substantially better hyperparameter configuration than a grid or random search (using the same budget):
+
+```{r, output=FALSE}
+instance = ti(
+  task = tsk("sonar"),
+  learner = learner,
+  resampling = rsmp("cv", folds = 3),
+  measures = msr("classif.ce"),
+  terminator = trm("evals", n_evals = 25)
+)
+tuner = tnr("grid_search", resolution = 5)
+tuner$optimize(instance)
+```
+
+```{r}
+instance$result
+```
+
+```{r}
+instance = ti(
+  task = tsk("sonar"),
+  learner = learner,
+  resampling = rsmp("cv", folds = 3),
+  measures = msr("classif.ce"),
+  terminator = trm("evals", n_evals = 25)
+)
+tuner = tnr("random_search", batch_size = 25)
+tuner$optimize(instance)
+```
+
+```{r, output=FALSE}
+instance$result
+```
+
+## Multi-Objective Bayesian Optimization {#sec-multi-objective-bayesian-optimization}
```

Comment:
On first read unclear what this section adds that hasn't been covered in 12.4 and C4